### PR TITLE
Additional queries validation

### PR DESCRIPTION
**Pull request by egor-ryashin**
see https://github.com/rilldata/rill/issues/2633

### CODE REVIEW

File: proto/rill/runtime/v1/queries.proto [hunk 1]
```diff
@@ -557,17 +557,17 @@ message ColumnCardinalityResponse {
...
-  string table_name = 2;
-  repeated BasicMeasure measures = 3;
-  string timestamp_column_name = 4;
+  string table_name = 2 [(validate.rules).string.min_len = 1];
+  repeated BasicMeasure measures = 3 [(validate.rules).repeated.min_items = 1];
+  string timestamp_column_name = 4 [(validate.rules).string.min_len = 1];
   TimeSeriesTimeRange time_range = 5;
-  int32 pixels = 7;
-  int32 sample_size = 8;
+  int32 pixels = 7 [(validate.rules).int32.gt = 0];
```
Comment (begelundmuller) [inline proto/rill/runtime/v1/queries.proto:564]: Doesn't the API support 0 for this?

File: runtime/server/queries_metrics.go [hunk 2]
```diff
@@ -224,3 +244,10 @@ func validateInlineMeasures(ms []*runtimev1.InlineMeasure) error {
 	}
 	return nil
 }
+
+func validateMeasures(ms []*runtimev1.InlineMeasure, measures []string) error {
+	if len(ms)+len(measures) == 0 {
+		return fmt.Errorf("no measures specified")
+	}
+	return nil
+}
```
Comment (begelundmuller) [inline runtime/server/queries_metrics.go:253]: I think you can just use proto validations – only the `measures` slice defines what measures are returned, the inline measures I believe just inject extra definitions

**Review by begelundmuller — CHANGES_REQUESTED** [review]

File: proto/rill/runtime/v1/queries.proto (hunk 1 above)
Comment (egor-ryashin) [reply proto/rill/runtime/v1/queries.proto]: yes, good catch

**Review by begelundmuller — CHANGES_REQUESTED** [review]
1. Needs to re-generate the proto files for changes to take effect
2. Please do `make cli` and do an experience pass on `./rill start` with a simple project?Just to make sure the UI doesn’t depend on making funny queries

**Review by begelundmuller — APPROVED** [review]

**Pull request merged**